### PR TITLE
Step 5. Data models defined

### PR DESCRIPTION
**Pull request by akimov228aleksei**
Models have been created.

### CODE REVIEW

File: src/management/backend/DAO.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""The file contains all the available methods
```
Comment (jimmyhollywood) [inline src/management/backend/DAO.py:1]: At this step we do not need DAO layer

File: src/management/backend/models/department.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class Department(models.Model):
+    """Class containing fields and methods of the model"""
+
+    title = models.CharField(max_length=100, unique=True)
+    description = models.TextField(blank=True)
+    time_create = models.DateTimeField(auto_now_add=True)
+    time_update = models.DateTimeField(auto_now=True)
+
+    @property
+    def amount(self):
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:13]: Please, use more obvious name for this method

File: src/management/backend/models/department.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        """A function that counts the number of employees in each department"""
+
+        from backend.DAO import EmployeeDAO
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:16]: Model mustn't depend on DAO layer. If you need to count number of employees then use sql count statement (see in django docs) or any other option

File: src/management/backend/models/department.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        return len(EmployeeDAO.filter_department(value=self.pk))
+
+    @property
+    def average_salary(self):
+        """A function that calculates the average salary of each department"""
+
+        from backend.DAO import EmployeeDAO
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:23]: The same. One more - imports in the middle of the code - it is not a good idea

**Review by jimmyhollywood — CHANGES_REQUESTED** [review]

File: src/management/backend/DAO.py (hunk 1 above)
Comment (akimov228aleksei) [reply src/management/backend/DAO.py]: Deleted.

File: src/management/backend/models/department.py (hunk 2 above)
Comment (akimov228aleksei) [reply src/management/backend/models/department.py]: Renamed.

File: src/management/backend/models/department.py (hunk 3 above)
Comment (akimov228aleksei) [reply src/management/backend/models/department.py]: Fixed.

File: src/management/backend/models/department.py (hunk 4 above)
Comment (akimov228aleksei) [reply src/management/backend/models/department.py]: Fixed.

File: src/management/backend/models/department.py (hunk 3 above)
Comment (jimmyhollywood) [reply src/management/backend/models/department.py]: Not yet. Please read closely documentation fro models in django and aggregation

Comment (akimov228aleksei) [reply src/management/backend/models/department.py]: I hope I've corrected it now

File: src/management/backend/models/department.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    title = models.CharField(max_length=100, unique=True)
+    description = models.TextField(blank=True)
+    time_create = models.DateTimeField(auto_now_add=True)
+    time_update = models.DateTimeField(auto_now=True)
+
+    @property
+    def total_count_employee(self):
+        """A function that counts the number of employees in each department"""
+
+        count = Employee.objects.values('department_id').annotate(employee=Count('id')).filter(department_id=self.pk)
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:18]: Look into the https://docs.djangoproject.com/en/4.0/topics/db/aggregation/#generating-aggregates-for-each-item-in-a-queryset
Maybe it is possible to simplify the request

**Review by jimmyhollywood — CHANGES_REQUESTED** [review]

File: src/management/backend/models/department.py (hunk 5 above)
Comment (akimov228aleksei) [reply src/management/backend/models/department.py]: Simplified this query

File: src/management/backend/models/department.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    title = models.CharField(max_length=100, unique=True)
+    description = models.TextField(blank=True)
+    time_create = models.DateTimeField(auto_now_add=True)
+    time_update = models.DateTimeField(auto_now=True)
+
+    @property
+    def total_count_employee(self):
+        """A function that counts the number of employees in each department"""
+
+        count = Department.objects.annotate(Count('employee')).filter(pk=self.pk)
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:18]: Why do you need here filtering? There is a pretty simply way to get number of foreign key references - https://docs.djangoproject.com/en/4.0/ref/models/querysets/#annotate

File: src/management/backend/models/department.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+        return count[0].employee__count if count else 0
+
+    @property
+    def average_salary(self):
+        """A function that calculates the average salary of each department"""
+
+        salary = Employee.objects.values('department_id').annotate(salary=Avg('salary')).filter(department_id=self.pk)
```
Comment (jimmyhollywood) [inline src/management/backend/models/department.py:26]: Let's leave this as is but you definitely have to refactor this part. Let's maybe create an issue in github Issues tab for this and will return to that later

**Review by jimmyhollywood — APPROVED** [review]
Fix count query for department according to link I added and then merge without my approval

**Pull request merged**